File: buerguer-pig/HamburgesasEIngredientes-Java-IvanRC/src/main/java/repository/hamburgesa/HamburgesaRepositoryImpl.java
package repository.hamburgesa;

import models.Hamburgesa;
import storage.hamburgesa.HamburgesaStorageService;

import java.io.IOException;
import java.util.ArrayList;
import java.util.List;
import java.util.UUID;

public class HamburgesaRepositoryImpl implements HamburgesaRepository{

    private final HamburgesaStorageService storage;

    public HamburgesaRepositoryImpl(HamburgesaStorageService storage) {
        this.storage = storage;
    }

    private ArrayList<Hamburgesa> hamburgesas = new ArrayList<>();

    @Override
    public List<Hamburgesa> getAll() throws Exception {
        loadAll();
        return hamburgesas;
    }

    @Override
    public Hamburgesa getById(UUID id) throws Exception {
        loadAll();
        return hamburgesas.stream().filter(hamburesa -> hamburesa.getId().equals(id)).findFirst().orElse(null);
    }

    @Override
    public Hamburgesa add(Hamburgesa entity) throws Exception {
        Hamburgesa hamburgesa = getById(entity.getId());
        if(hamburgesa == null){
            hamburgesas.add(entity);
            saveAll();
            return entity;
        }
        return null;
    }

    @Override
    public Hamburgesa update(Hamburgesa entity) {
        return null;
    }

    @Override
    public Hamburgesa remove(UUID id) {
        return null;
    }

    @Override
    public List<Hamburgesa> getByName(String name) {
        return null;
    }


    private void loadAll() throws Exception {
        hamburgesas.clear();
        storage.loadAll().forEach(hamburgesa ->
                hamburgesas.add(hamburgesa)
        );
    }

    private void saveAll() throws Exception {
        storage.saveAll(hamburgesas);
    }
}

File: buerguer-pig/HamburgesasEIngredientes-Kotlin-IvanRC/src/main/kotlin/repository/Hamburger/HamburgesaRepositoryImpl.kt
package repository.Hamburger

import model.Hamburgesa
import mu.KotlinLogging
import storageService.Hamburger.HamburgesaStorageService
import java.util.*

class HamburgesaRepositoryImpl(
    private val storage: HamburgesaStorageService
): HamburgesaRepository {

    private val logger = KotlinLogging.logger {}
    private var hamburgesas = mutableListOf<Hamburgesa>()

    override fun findByName(name: String): List<Hamburgesa> {
        logger.debug { "Se buscan hamburgesas según su nombre." }
        TODO("Not yet implemented")
    }

    override fun findById(id: UUID): Hamburgesa? {
        logger.debug { "Se busca una hamburgesa según su id." }
        loadAll()
        return hamburgesas.find { it.id == id }
    }

    override fun getAll(): List<Hamburgesa> {
        logger.debug { "Se consiguen hamburgesas." }
        loadAll()
        return hamburgesas.toList()
    }

    override fun add(entity: Hamburgesa): Hamburgesa? {
        logger.debug { "Se añade una hamburgesa." }
        (findById(entity.id))?:run{
            hamburgesas.add(entity)
            saveAll()
            return entity
        }
        return null
    }

    override fun delete(id: UUID): Hamburgesa? {
        logger.debug { "Se elimina una hamburgesa." }
        TODO("Not yet implemented")
    }

    override fun update(entity: Hamburgesa): Hamburgesa? {
        logger.debug { "Se actualiza una hamburgesa." }
        TODO("Not yet implemented")
    }

    fun saveAll(){
        logger.debug { "Se guargan las hamburgesas en el storage." }
        storage.saveAll(hamburgesas)
    }

    fun loadAll(){
        logger.debug { "Se cargan las hamburgesas en el storage." }
        hamburgesas = storage.loadAll().toMutableList()
    }
}